src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Source/AnalyzeDiagnostics.cs

```diff
@@ -0,0 +1,139 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.DotNet.UpgradeAssistant.Analysis;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Steps.Source
+{
+    public class AnalyzeDiagnostics : IAnalyzeResultProvider
+    {
+        private readonly IEnumerable<DiagnosticAnalyzer> _allAnalyzers;
+        private readonly ImmutableArray<AdditionalText> _additionalTexts;
+        private readonly ILogger _logger;
+
+        internal IEnumerable<Diagnostic> Diagnostics { get; set; } = Enumerable.Empty<Diagnostic>();
```

Comment:
Why is this internal? What's it used for? Does it need to maintain state of the result provider?

Reply:
removed